backend/src/main/java/com/github/redstylzz/backend/service/DepositService.java

```diff
@@ -0,0 +1,86 @@
+package com.github.redstylzz.backend.service;
+
+import com.github.redstylzz.backend.exception.DepositAlreadyExistException;
+import com.github.redstylzz.backend.exception.DepositDoesNotExistException;
+import com.github.redstylzz.backend.model.Deposit;
+import com.github.redstylzz.backend.model.dto.DepositDTO;
+import com.github.redstylzz.backend.repository.IDepositRepository;
+import lombok.RequiredArgsConstructor;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.springframework.stereotype.Service;
+
+import java.time.*;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class DepositService {
+    private static final Log LOG = LogFactory.getLog(DepositService.class);
+
+    private final IDepositRepository repository;
+
+    private List<DepositDTO> getDepositsAsDTO(String userId) {
+        return repository.getAllByUserId(userId).stream().map(Deposit::mapDepositToDTO).toList();
+    }
+
+    private boolean isValidDeposit(Deposit deposit) {
+        return (deposit.getDescription() != null &&
```

Comment:
Is an empty string a valid description?

you could have a look at javax.validation